pkg/distributor/distributor.go

```diff
@@ -1237,6 +1244,14 @@ func (d *Distributor) push(ctx context.Context, pushReq *push.Request) (*mimirpb
 	return &mimirpb.WriteResponse{}, nil
 }
 
+func (d *Distributor) getTokensForSeries(userID string, series []mimirpb.PreallocTimeseries) []uint32 {
+	result := make([]uint32, 0, len(series))
```

Comment:
in many cases the list of ephemeral time series will be empty, should we shortcut in this case and just return `nil` instead of allocating an empty slice?

Reply:
Done in https://github.com/grafana/mimir/pull/3889/commits/8918642834bbde08f159969b5676039e5e5f97b6.